RFC-0003-arrow-flight-connector.md

```diff
@@ -0,0 +1,70 @@
+# **RFC0003 for Presto**
+
+## Arrow Flight connector template in Presto
+
+Proposers
+
+* Elbin Pallimalil
+
+## [Related Issues]
+
+## Summary
+
+Create a base Arrow Flight module in Presto which can be extended to create an Arrow Flight connector.
+
+## Background
+
+Apache Arrow Flight is a new general-purpose client-server framework that simplifies high performance transport of large datasets over network interfaces.
+
+Presto currently supports different datasources through connectors for each type of datasource. The connectors could be using JDBC or some other protocol for datasource connection. If we use Flight client to connect to the different datasources we can improve the performance of the datasource connection. Also we would only need to develop one Flight client that can be used across different datasources instead of developing individual connectors for each type of datasource.
```

Comment:
Are there any preliminary estimates of the performance improvement? What exactly does "performance of the datasource connection" refer to?

Reply:
We have an initial version of Arrow Flight connector running. Response time for SELECT query seem to be about 30 to 60 percent faster than JDBC, depending on the number of rows in the table.